src/v/cloud_topics/level_one/metastore/leader_router.cc

```diff
@@ -265,6 +265,124 @@ leader_router::process(req_t req, bool local_only) {
     co_return resp_t{.ec = rpc::errc::not_leader};
 }
 
+template<auto LocalFunc, auto RemoteFunc, typename req_t>
+requires requires(
+  cloud_topics::l1::leader_router f, const model::ntp& ntp, req_t req) {
+    (f.*LocalFunc)(std::move(req), ntp, ss::shard_id{0});
+    request_is_vectorized<req_t>;
+}
+ss::future<typename req_t::resp_t>
+leader_router::process_vectorized(req_t req, bool local_only) {
+    static const auto req_name = ss::pretty_type_name(typeid(req_t));
```

Comment:
nit: not suggesting a change, but I found this tough to review without bits of the next commit for context. Maybe would've helped to bring in the changes to replicated_metastore into the same commit, without extending the metastore interface. Or giving a bit more depth about the new "vectorized" rpc types